### PR TITLE
Re-Land: Add focus nodes, hover, and shortcuts to switches, checkboxes, and radio buttons.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This re-lands the change that adds focus nodes, hover, and shortcuts to switches, checkboxes, and radio buttons. (#43213)

No changes from original, except for finding the right `RenderBox` in `dev/integration_tests/android_semantics_testing/test_driver/main_test.dart`.

## Breaking Change

- [X] No, this is *not* a breaking change.

### CODE REVIEW

File: dev/integration_tests/android_semantics_testing/test_driver/main_test.dart [hunk 1]
```diff
@@ -198,8 +198,16 @@ void main() {
       });
 
       test('Checkbox has correct Android semantics', () async {
+        Future<AndroidSemanticsNode> getCheckboxSemantics(String key) async {
+          return getSemantics(
+            find.descendant(
+              of: find.byValueKey(key),
+              matching: find.byType('_CheckboxRenderObjectWidget'),
```
Comment (darrenaustin) [inline dev/integration_tests/android_semantics_testing/test_driver/main_test.dart]: Why does this need to be against `_CheckboxRenderObjectWidget` and not just the key?

File: packages/flutter/test/material/checkbox_test.dart [hunk 2]
```diff
@@ -67,11 +67,12 @@ void main() {
       ),
     ));
 
-    expect(tester.getSemantics(find.byType(Checkbox)), matchesSemantics(
+    expect(tester.getSemantics(find.byType(Focus)), matchesSemantics(
```
Comment (darrenaustin) [inline packages/flutter/test/material/checkbox_test.dart]: Why the change from `Checkbox` to `Focus` here?

**Review by darrenaustin — APPROVED** [review]
There's a lot here, and I might have missed something, but this LGTM.

File: dev/integration_tests/android_semantics_testing/test_driver/main_test.dart (hunk 1 above)
Comment (gspencergoog) [reply dev/integration_tests/android_semantics_testing/test_driver/main_test.dart]: Because the key corresponds to a `RenderMouseRegion` now that I've added the `MouseRegion`, so we have to dig down to find the render object for the checkbox which adds the semantics.

File: packages/flutter/test/material/checkbox_test.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter/test/material/checkbox_test.dart]: Because that gets us down past the `RenderMouseRegion` so that it gets the `_CheckboxRenderObjectWidget` render object to query semantics from.  Since `Focus` doesn't have a render object associated with it, it defers to the `_CheckboxRenderObjectWidget`.

**Comment by yjbanov** [comment]
This PR seems to have broken web tests on master.
